changeloggenutils/minor_release.go

```diff
@@ -0,0 +1,353 @@
+package changelogdocutils
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/google/go-github/v32/github"
+	"github.com/solo-io/go-utils/githubutils"
+	. "github.com/solo-io/go-utils/versionutils"
+	"github.com/yuin/goldmark"
+	"github.com/yuin/goldmark/ast"
+	"github.com/yuin/goldmark/text"
+	"math"
+	"sort"
+	"strings"
+)
+
+type MinorReleaseGroupedChangelogGenerator struct {
+	Client *github.Client
+	opts   Options
+}
+
+/**
+ Groups releases by their minor version:
+ v1.8
+ - v1.8.2
+ - v1.8.1
+ v1.7
+ - v1.7.1
+ - v1.7.0-beta9
+ ...
+ */
+func NewMinorReleaseGroupedChangelogGenerator(opts Options, client *github.Client) *MinorReleaseGroupedChangelogGenerator {
+	if opts.NumVersions == 0{
+		opts.NumVersions = math.MaxInt64
```

Comment:
Can we change the default to 1? I'm just afraid of someone forgetting to set the NumVersions option, and then we pull down tons of release notes

Reply:
By default, we want to pull down all release notes, as is currently being done in our changelogs. If we set default to 1, that will only pull down one release. `NumVersions` is maybe for later stage products with thousands of releases.